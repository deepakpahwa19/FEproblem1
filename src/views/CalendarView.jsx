import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

export const CalendarView = ({ month, monthName }) => {
    const [today, setTodayDate] = useState(new Date());

    useEffect(() => {
        setInterval(() => {
            if (today !== new Date()) setTodayDate(new Date());
        }, 1000);
    }, []);

    return (
        <CalendarStyle>
            <Header>
                <h4>April 2021</h4>
                <div>icons</div>
            </Header>
            <div>
                <Table>
                    <Thead>
                        <TR>
                            <TD>S</TD>
                            <TD>M</TD>
                            <TD>T</TD>
                            <TD>W</TD>
                            <TD>T</TD>
                            <TD>F</TD>
                            <TD>S</TD>
                        </TR>
                    </Thead>
                    <tbody>
                        {month.map(week => (
                            <TR>
                                {week.map(date => (
                                    <TD currentDay={today.getDate() === date}>{date}</TD> //
                                ))}
                            </TR>
                        ))}
                    </tbody>
                </Table>
            </div>
        </CalendarStyle>
    );
};

const CalendarStyle = styled.div`
    width: 400px;
    height: 400px;
    background-color: #e9ece2;
    border: 1px solid grey;
    margin: 10px auto;
    padding: 20px;
`;

const Header = styled.div`
    height: 30px;
    margin-bottom: 20px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
`;

const Table = styled.table`
    width: 100%;
`;

const Thead = styled.thead``;

const TR = styled.tr`
    width: 100%;
    display: flex;
    margin: 10px 0 20px;
    justify-content: space-between;
`;

const TD = styled.td`
    width: 55px;
    background-color=${props => (props.currentDay ? 'red' : '')}
`;
