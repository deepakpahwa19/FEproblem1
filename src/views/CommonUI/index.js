export { H4, H1, H3 } from './HeaderGroup';
export { RequiredElement } from './RequiredElement';
export { Button } from './Button';
