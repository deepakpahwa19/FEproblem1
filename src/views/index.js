export { DropDownView } from './DropdownView/DropDownView';
export { RadioButtonView } from './RadioButtonView/RadioButtonView';
export { FlexContainer } from './StyledComponents/FlexContainer';
export { JourneyCardView } from './JourneyCardView/JourneyCardView';
export { H1, H4, H3, RequiredElement } from './CommonUI';
export { FooterView } from './Footer/Footer';
export { Spinner } from './SpinnerView/Spinner';
export { Button } from './CommonUI/ButtonView';
export { ErrorMessage } from './CommonUI/ErrorMessage';
export { Backdrop } from './Backdrop/Backdrop';
