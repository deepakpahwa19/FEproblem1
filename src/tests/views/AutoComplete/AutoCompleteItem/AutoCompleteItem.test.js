import { shallow } from 'enzyme';
import { AutoCompleteItem } from '../../../../views';

describe('Verifying AutoCompleteItem in View', () => {
    let name = 'Bangalore';
    const onClickHandler = jest.fn();

    it('Verifying AutoCompleteItem default snapshot', () => {
        const wrapper = shallow(<AutoCompleteItem onSelectItem={onClickHandler} name={name} isHighlighted />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Verifying AutoCompleteItem click and name', () => {
        const wrapper = shallow(<AutoCompleteItem onSelectItem={onClickHandler} name={name} />);
        expect(wrapper.find('p').text()).toEqual(name);

        wrapper.simulate('click');
        wrapper.setProps({ name: 'Delhi' });

        expect(wrapper.find('p').text()).toEqual('Delhi');
        expect(wrapper).toMatchSnapshot();
    });
});
