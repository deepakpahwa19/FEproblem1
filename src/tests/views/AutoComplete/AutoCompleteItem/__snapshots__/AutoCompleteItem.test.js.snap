// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verifying AutoCompleteItem in View Verifying AutoCompleteItem click and name 1`] = `ShallowWrapper {}`;

exports[`Verifying AutoCompleteItem in View Verifying AutoCompleteItem default snapshot 1`] = `ShallowWrapper {}`;
